strict-mvar/src/Control/Concurrent/Class/MonadMVar/Strict/Checked.hs

```diff
@@ -0,0 +1,188 @@
+{-# LANGUAGE BangPatterns  #-}
+{-# LANGUAGE CPP           #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | This module corresponds to 'Control.Concurrent.MVar' in "base" package
+--
+module Control.Concurrent.Class.MonadMVar.Strict.Checked
```

Comment:
I realise that the modules in `strict-mvar` live in a different module hierarchy than the lazy MVars, which come from `Control.Monad.Class.MonadMVar`. We should resolve this discrepancy, either by renaming the modules in `strict-mvar` or by renaming the `Control.Monad.Class.MonadMVar` module.

Reply:
`base` is using `Control.Concurrent.MVar` so it makes sense to fix this in `io-classes`.

Reply:
https://github.com/input-output-hk/io-sim/pull/88 makes the change in `io-classes`